tests/unit/flow/test_flow.py

```diff
@@ -559,3 +560,31 @@ def test_flow_arbitrary_needs():
 
     with f:
         f.index_lines(['abc', 'def'])
+
+
+def test_flow_needs_all():
+    f = (Flow().add(name='p1', needs='gateway')
+         .needs_all(name='r1'))
+    assert f._pod_nodes['r1'].needs == {'p1'}
+
+    f = (Flow().add(name='p1', needs='gateway')
+         .add(name='p2', needs='gateway')
+         .add(name='p3', needs='gateway')
+         .needs(needs=['p1', 'p2'], name='r1')
+         .needs_all(name='r2'))
+    assert f._pod_nodes['r2'].needs == {'p3', 'r1'}
+
+    with f:
+        f.index_ndarray(np.random.random([10, 10]))
```

Comment:
```suggestion

```

Reply:
Thanks for your comment!
Those were added to address  (https://github.com/jina-ai/jina/pull/1288#pullrequestreview-531216135)  requested  by @maximilianwerk or did I misunderstood the requirement? Feel free to let me know whether  keep them or replace those tests with better ones. Thanks!

Reply:
I had a short chat with @hanxiao and there we agreed, that a short test with data flowing through the flow would be beneficial in order to test the empty pods.

Reply:
yes. i think its all good, merge shall we?